cpp/src/arrow/ipc/reader.cc

```diff
@@ -1000,6 +1049,13 @@ static Future<std::shared_ptr<Message>> ReadMessageFromBlockAsync(
                           io_context);
 }
 
+static Result<std::unique_ptr<Message>> ReadMessageFromCached(
+    std::shared_ptr<Buffer> cached_metadata, std::shared_ptr<Buffer> cached_data) {
+  ARROW_ASSIGN_OR_RAISE(auto message,
+                        ReadMessage(std::move(cached_metadata), std::move(cached_data)));
+  return std::move(message);
+}
+
```

Comment:
I kind of question having this method over just calling ReadMessage directly.

Reply:
Good point.  I inlined it.